rust/src/detect/xbits.rs

```diff
@@ -0,0 +1,224 @@
+use std::ffi::{CStr, CString};
+use std::os::raw::{c_char, c_void};
+
+pub const DETECT_XBITS_CMD_SET: u8 = 0;
+pub const DETECT_XBITS_CMD_TOGGLE: u8 = 1;
+pub const DETECT_XBITS_CMD_UNSET: u8 = 2;
+pub const DETECT_XBITS_CMD_ISNOTSET: u8 = 3;
+pub const DETECT_XBITS_CMD_ISSET: u8 = 4;
+pub const DETECT_XBITS_CMD_NOALERT: u8 = 5;
+pub const DETECT_XBITS_CMD_MAX: u8 = 6;
+pub const DETECT_XBITS_TRACK_IPSRC: u8 = 0;
+pub const DETECT_XBITS_TRACK_IPDST: u8 = 1;
+pub const DETECT_XBITS_TRACK_IPPAIR: u8 = 2;
+pub const DETECT_XBITS_TRACK_FLOW: u8 = 3;
+pub const DETECT_XBITS_EXPIRE_DEFAULT: u32 = 30;
```

Comment:
Is there a way to de duplicate these? I remember @catenacyber asking for something like that a while ago. @catenacyber Did you manage to?

Reply:
These are already being export to C with `cbindgen`, for example:
```
$ grep DETECT_XBITS_CMD_TOGGLE gen/rust-bindings.h 
#define DETECT_XBITS_CMD_TOGGLE 1
```

We're not seeing a conflic on the C side as there is no code that includes `detect-xbits.h` and `rust-bindings.h` yet.  You could try including `rust.h` to pull these in and remove them from `detect-xbits.h` and it might work out OK.  There are still some header ordering issues where you can end up with headers that are required before rust.h, these can't depend on stuff defined in Rust. This prevents us from moving some constants into Rust like the ALPROTO_ values.